Extending/Packages/Good-practice/index.md

```diff
@@ -0,0 +1,161 @@
+
+The default delivery method for files to the App_Plugins folder is via a `.targets` file within a package, this means when a website is built the files in this folder are copied over from the nuget cache, and any changes a user might have made will be lost. Equally if the user performs a `dotnet clean` on a solution all files in the App_Plugins folder will likely be deleted. 
+
+If you have files that you expect users of your package to alter you should not place them in the `App_Plugins` folder.
+
+## Views
+Views are used to render content on the front end of a website. If your plugin provides a way for the user to present the content publicly you should copy these files to the views folder. 
+
+As the files will still be copied during build you should ensure your target file doesn't overwrite newer or altered files and doesn't delete files on clean. 
+
+_This method isn't perfect and if you do have a package that is heavy on views you might consider building a dotnet template or Razor class library to contain the views so they don't pollute your user's solutions._ 
+
+## License files
+Umbraco products store their licenses in `/umbraco/Licences` and it is considered a good location for third party packages that require license files to also store their licence files there. 
+
+_The default .gitignore for Umbraco templates will include any files in this folder in a repo (while ignoring most of the rest of the Umbraco folder)_
+
+:::note
+The `/umbraco/Licenses` folder does not exist on a fresh installation of Umbraco so you might need to create it before you save your license file to this folder. 
+:::
+
+## Operating system considerations
+Umbraco 9 is a .NET Core application and it can be ran on multiple operating systems (e.g Windows, Linux). When developing packages there are a few things you should be aware of so your package can run on all possible operating systems. 
+
+### Case sensitivity 
+The Linux file system is case sensitive, so `App_Plugins/myPackage` is a different location to `app_plugins/MyPACKAGE`. When building your package you should ensure that you always refer to folders and paths in a consistent way. 
+
+:::tip
+You can adjust the case sensitivity of a Windows folder by running a command against the folder. 
+
+``` bash
+fsutil.exe file queryCaseSensitiveInfo <path>
+```
+:::
+
+### Well known folders and case sensitivity  
+Some folders within Umbraco will already exist for all installations, if you access these folders you need to be aware of the case used to ensure you end up in the right place: 
+
+| Folder | Note | 
+|-|-|
+| /App_Plugins | Uppercase `A` and `P` |
+| /App_Plugins/[Ll]ang | Uppercase `L` but in later version of Umbraco 9.3+ can be upper or lowercase `L` |
+| /Views | Uppercase `V` | 
+| /umbraco/Licenses | Lowercase `u` uppercase `L` |
+| /config | Lowercase `c` |
```

Comment:
This isn't used in v9 anymore (and the `umbraco\config\lang` folder should only include project specific/user language overrides, so shouldn't be something for package authors to deal with).

Reply:
i think the caseness of these folders is worth noting, and if you want to make a package that works on all v9s you need to know that pre v9.3 the case thing was an issue. (again if we make this a v10 doc, we can simplify)